Library/Homebrew/version.rb

```diff
@@ -379,6 +379,21 @@ def self._parse(spec, detected_from_url:)
   end
   private_class_method :_parse
 
+  DOT_OPTIONAL = /(?:\d+(?:\.\d+)*)/.source.freeze
+  private_constant :DOT_OPTIONAL
+
+  DOT_REQUIRED = /(?:\d+(?:\.\d+)+)/.source.freeze
+  private_constant :DOT_REQUIRED
+
+  MINOR_OR_PATCH = /(?:\d+(?:\.\d+){1,2})/.source.freeze
+  private_constant :MINOR_OR_PATCH
+
+  BIN_SUFFIX = /(?:[._-](?i:bin|dist|stable|src|sources?|final|full))/.source.freeze
+  private_constant :BIN_SUFFIX
+
+  ALPHA_SUFFIX = /(?:[._-]?(?i:alpha|beta|pre|rc)\.?\d{,2})/.source.freeze
```

Comment:
```suggestion
  PRERELEASE_SUFFIX = /(?:[._-]?(?i:alpha|beta|pre|rc)\.?\d{,2})/.source.freeze
```

Since this regex is meant for detecting suffixes for more pre-release versions than just alpha builds.

Reply:
I actually originally had `PREREL_SUFFIX` but I decided to go with the first value in `(?i:)` for simplicity. Personally I don't think the name matters much, especially since this is a private constant in Homebrew internals, but we can always rename it in a follow-up PR. I might rename it when I work on adding constants for livecheck